### ISSUE TITLE
False positive E0277 "trait bound `Sized` not satisfied"

### ISSUE THREAD

**Issue by rrevenantt** [intellij-rust/intellij-rust#2530]
False positive E0277 in rust standard library in iterator.rs:

![image](https://user-images.githubusercontent.com/8453142/40023777-d6e9ffd8-57d4-11e8-8b50-c659080398d1.png)
\
\
intellij-rust from master
Intellij IDEA 2018.1.1
rust 1.26 stable
Manjaro Linux

**Comment by ArtsiomCh** [comment]
Same issue for mutex.rs as well:
![image](https://user-images.githubusercontent.com/14268320/54303842-47006880-459a-11e9-8c67-d9dd9f13f13b.png)

IntelliJ IDEA 2019.1 EAP (Community Edition)
Build #IC-191.5849.21, built on February 27, 2019
JRE: 1.8.0_202-release-1483-b31 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 4.4.0-142-generic
Rust plugin v0.2.93.2117-191

**Comment by CAD97** [comment]
cc #2783, the apparent cause

minimal cases:

- impl header, where bound

```rust
struct Wrapper<T: ?Sized>(T);

impl<T: ?Sized> Wrapper<T> {
    fn into_inner(self) -> T
    where T: Sized
    { self.0 }
}
```

- fn header, fn header

```rust
fn foo<T: ?Sized + Sized>(t: T) -> T { t }
```
